zh-learn-cli/src/main/java/com/zhlearn/cli/ParsePlecoCommand.java

```diff
@@ -105,32 +104,42 @@ public void run() {
             // Create dictionary for any dictionary-based providers
             PlecoExportDictionary dictionary = new PlecoExportDictionary(entries);
 
-            // Register dictionary-backed providers so users can opt-in via --*-provider flags
-            var registry = parent.getProviderRegistry();
-            registry.registerDefinitionProvider(new DictionaryDefinitionProvider(dictionary));
-            registry.registerPinyinProvider(new DictionaryPinyinProvider(dictionary));
-            
+            // Note: Dictionary providers are no longer dynamically registered
+            // They are created at startup in MainCommand
+
+            // Set parse-pleco specific defaults first
+            String effectiveDefinitionProvider = definitionProvider != null ? definitionProvider : "pleco-export";
```

Comment:
instead of this complex logic, can't I just use the picocli to set the defaults? this would be much cleaner